### PR TITLE
Improvement to process.py logging

### PR DESCRIPTION
**Pull request by rkoumis**
Log the image name as well as the process ID; added tests. 
Also, restore the process is_alive check, disabled 4 years ago.

Credit: @nbargnesi

Sample log entries:
```
2024-03-10 12:12:13,411 [lib.api.process] DEBUG: getting exit code for <Process 17492 aspnet_compiler.exe>
2024-03-10 12:12:13,411 [lib.api.process] DEBUG: <Process 17492 aspnet_compiler.exe> exit code is 259
2024-03-10 12:12:13,411 [lib.api.process] DEBUG: <Process 17492 aspnet_compiler.exe> is still active
```

### CODE REVIEW

**Comment by doomedraven** [comment]
LGTM, what about you @kevoreilly ?

**Comment by kevoreilly** [comment]
If we can quiten these messages down to an 'acceptable' level then I think it should be ok.

**Comment by rkoumis** [comment]
I've commented out the verbose / spammy logging, @kevoreilly @doomedraven 

I squashed the commits back down to one, and now I realize I probably shouldn't have, sorry. Anyhow, that's what I did. Let me know if any questions.

**Comment by doomedraven** [comment]
im not having time right now to test, was you able to test @kevoreilly latest commit? if no i will do that in 1-2 days

**Comment by kevoreilly** [comment]
Yep I'll test it tomorrow

**Comment by rkoumis** [comment]
no rush but please take another look when you get a chance

**Comment by kevoreilly** [comment]
I've been running this on my main server for the last few days - I just need to check the logs.

**Comment by kevoreilly** [comment]
I wonder if this line is ok:

```
log.debug("Failed getting exit code for %s", self)
```

Is ``self`` not gonna bomb as not a string type?

File: analyzer/windows/lib/api/process.py [hunk 1]
```diff
@@ -642,6 +682,11 @@ def upload_memdump(self):
...
             log.error(file_path)
-        log.info("Memory dump of process %d uploaded", self.pid)
+        log.info("memory dump of %s uploaded", self)
 
         return True
+
+    def __str__(self):
+        """Get a string representation of this process."""
+        image_name = self.get_image_name() or "???"
+        return f"<{self.__class__.__name__} {self.pid} {image_name}>"
```
Comment (rkoumis) [inline analyzer/windows/lib/api/process.py:692]: This method allows us to do `log.info("%s", self)`

Comment (kevoreilly) [reply analyzer/windows/lib/api/process.py:692]: Ah ok cool - sorry to be a pain!

**Comment by rkoumis** [comment]
The new `__str__()` method will take care of converting a `Process` to a string. That's how come in your first round of testing you saw messages like
```
DEBUG: <Process 4364 explorer.exe> is still active
```

```python
    def __str__(self):
        """Get a string representation of this process."""
        image_name = self.get_image_name() or "???"
        return f"<{self.__class__.__name__} {self.pid} {image_name}>"
```

**Comment by kevoreilly** [comment]
Thank you!

**Pull request merged**